tpot2/individual_representations/graph_pipeline_individual/individual.py

```diff
@@ -547,15 +559,24 @@ def _mutate_hyperparameters(self):
         '''
         sorted_nodes_list = list(self.graph.nodes)
         random.shuffle(sorted_nodes_list) 
+        completed_one = False
         for node in sorted_nodes_list:
             if isinstance(node,GraphIndividual):
                 continue
             if isinstance(self.select_config_dict(node)[node.method_class], dict):
                 continue
-            node.hyperparameters = self.select_config_dict(node)[node.method_class](config.hyperparametersuggestor) 
-            
+
+            if not completed_one:
```

Comment:
Would the `else` part of this `if` ever get hit, given that completed_one is always set to False at the start?

Maybe just use the `hyper_node_probability`?

Reply:
good catch, just fixed it. 

The completed_one is to guarantee that at least one node has its hyperparameters mutated

Reply:
The `else` would still not get hit because of the `return True` statement's placement right? If the plan is just to mutate one node then this works fine, but if you want to have a probability of each node being mutated, this won't accomplish that.